docs-src/docs.md

```diff
@@ -0,0 +1,422 @@
+
+We only need the master and a prod branch for this demonstration - the idea is to have a branch per environment.
+
+Delete all other branches apart from master branch in your forked repo
+
+Setup a prod branch
+
+```
+git checkout -b prod
+git push origin prod
+git checkout master
+```
+
+### Create Jenkins image
+
+Project includes a [Dockerfile](https://github.com/flogei/ci-demo-walkthrough/blob/master/docs/Dockerfile) that creates a Docker image using Jenkins official Docker image as base. It also sets up required Jenkins plugins and copies job configurations to the image.
+
+```
+docker build --no-cache -t apigee/ci-demo-walkthrough .
+```
+
+A new docker image called apigee/ci-demo-walkthrough will be created. This takes a while. Take a note of the output, which is the public ssh key we are using in the next step.
+
+![image alt text](images/image_4.png)
+
+### Setup git ssh access
+
+This project uses ssh public/private key pairs in order to authenticate the jenkins user to git server. [Why?](https://github.com/flogei/ci-demo-walkthrough/blob/master/docs/faq.md#why-ssh-for-git-authentication)
+
+At the end of the image creation process, Dockerfile generates and prints out a new public key that you can setup to give jenkins push/pull access to your repo.
+
+If you pushed the project files to your GitHub account, then go to your repo and within Settings click Deploy Keys on the left-hand menu. Next Add deploy key Give it a meaningful name under Title, copy/paste the contents of the public key as a new key and enable Allow write access option. And finally hit Add key
+
+![image alt text](images/image_5.png)
+
+See [GitHub Deploy Keys Documentation](https://developer.github.com/guides/managing-deploy-keys/#deploy-keys)
+
+### Run image
+Positive
+: Note: remove any previous currency-v1 deployments from the Apigee org in order to demonstrate proxies created successfully with correct revision numbers.
+
+All parameters required to run Jenkins and Maven are supplied using environment variables. These variables are set directly to the docker container with the docker run command.
+
+There is a script provided to make this process easier which can be found [here](https://github.com/flogei/ci-demo-walkthrough/blob/master/docs/setup/run-image.sh). Modify this script to provide values specific to your setup and execute the script to run the image.
+
+Please note that setup/run-image.sh script is in source control. Copy this file to setup/.run-image.sh before making any changes to prevent sensitive data committed to this repo. Root .gitignore file contains a directive to ignore any changes to setup/.run-image.sh.
+
+Go to the cloned folder
+
+```
+cp ./setup/run-image.sh ./setup/.run-image.sh
+```
+
+Modify the file and plug in your environment data:
+
+```
+vim ./setup/.run-image.sh
+```
+
+
+You get that data from your Apigee Account. You can look it up by going to your helloworld proxy.
+
+![image alt text](images/image_6.png)
```

Comment:
having a screenshot for this is excellent.

is it possible that someone might not have a helloworld proxy? (they deleted it or the org is very very old so we never provisioned it)

in this case, maybe we should say "look at any proxy" or get them to look at the virtual host config instead?